GitUI/MessageBoxes.cs

```diff
@@ -127,6 +131,21 @@ public static bool ConfirmUpdateSubmodules(IWin32Window? owner)
             return result;
         }
 
+        public static bool ConfirmBranchCheckout(IWin32Window? owner)
+        {
+            TaskDialogPage page = new()
+            {
+                Text = Instance._confirmBranchCheckout.Text,
+                Heading = Instance._confirmCheckoutHeader.Text,
+                Caption = Instance._confirmCheckout.Text,
+                Icon = TaskDialogIcon.Warning,
```

Comment:
Why is it a warnings? We use the info icon in other dialogs.

Reply:
Please use `MessageBoxes.Confirm()`.

Feel free to make `MessageBoxes.Show()` use a `TaskDialog` instead of a `MessageBox` - rather in a separate PR though.